.github/workflows/ci-hsa-linux.yml

```diff
@@ -0,0 +1,354 @@
+              -c submodule."third_party/webgpu-headers".update=none \
+              -c submodule."third_party/musl".update=none \
+              -c submodule."third_party/stablehlo".update=none \
+              -c submodule."third_party/torch-mlir".update=none \
+              -c submodule."third_party/hip-build-deps".update=none \
+              submodule update --init --recursive --depth 1 --single-branch -j 10 third_party/iree
+
+      - name: Python deps
+        run: |
+          pip install "numpy<2" pyyaml "pybind11[global]==2.10.3" nanobind pytest
+
+      - name: Enable cache
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ env.CACHE_DIR }}
+          key:  ${{ env.CACHE_KEY }}
+          restore-keys: linux-build-test-cpp-
+
+      - name: Build packages
+        run: |
+          export cache_dir="${{ env.CACHE_DIR }}"
+          export CCACHE_COMPILERCHECK="string:$(clang --version)"
+          bash build_tools/build_llvm.sh
+          rm -rf llvm-build
+
+      - name: Create artifacts
+        if: ${{ !cancelled() }}
+        run: |
+          pushd third_party/iree/third_party/llvm-project && llvm_sha_short=$(git rev-parse --short HEAD) && popd
+          tar cf llvm-dist-linux-$llvm_sha_short.tar llvm-install
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        if: ${{ !cancelled() }}
+        with:
+          name: linux_x86_64_llvm_packages
+          path: llvm-dist-*.tar
+          if-no-files-found: warn
+
+      - name: Save cache
+        uses: actions/cache/save@v3
+        if: ${{ !cancelled() && github.event_name == 'push' && github.ref_name == 'main' }}
+        with:
+          path: ${{ env.CACHE_DIR }}
+          key: ${{ env.CACHE_KEY }}
+
+  build_hsa:
+    name: Build HSA (linux)
+    needs: [build_llvm]
+    runs-on: nod-ai-shared-cpubuilder-manylinux-x86_64
+    env:
+      CACHE_DIR: ${{ github.workspace }}/.container-cache
+      CACHE_KEY: linux-build-test-cpp-asserts-manylinux-v2-${{ format('{0}-{1}', github.ref_name, github.run_number) }}
+    steps:
+      - name: Configure local git mirrors
+        run: |
+          /gitmirror/scripts/trigger_update_mirrors.sh
+          /gitmirror/scripts/git_config.sh
+
+      - name: "Checking out repository"
+        env:
+          BRANCH_NAME: ${{ github.ref }}
+          REPO_ADDRESS: ${{ github.server_url }}/${{ github.repository }}
+        run: |
+          git init
+          git remote add origin $REPO_ADDRESS
+          git -c protocol.version=2 fetch --depth 1 origin $BRANCH_NAME
+          git reset --hard FETCH_HEAD
+          git submodule update --init --recursive --depth 1 --single-branch -j 10 third_party/ROCR-Runtime
+
+      - name: Install deps
+        run: |
+          dnf install -y almalinux-release-devel
+          yum install -y elfutils-libelf-devel p7zip p7zip-plugins \
+                         ncurses-compat-libs openssh vim-common
+
+      - name: Build and install libnuma
```

Comment:
Yes it's a RORr dep and we need to build it (instead of doing `apt install libnuma-dev`) because the runner isn't ubuntu it's almalinux and that thing isn't in any of the rhel/centos/whatever repos.

Reply:
Cool, thanks